sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -83,16 +80,12 @@ object ScalaNativePluginInternal {
   )
 
   lazy val scalaNativeConfigSettings: Seq[Setting[_]] = Seq(
-    nativeTarget := interceptBuildException {
-      val cwd   = nativeWorkdir.value.toPath
-      val clang = nativeClang.value.toPath
-      Discover.targetTriple(clang, cwd)
-    },
-    artifactPath in nativeLink := {
+    nativeLink / artifactPath := {
       crossTarget.value / (moduleName.value + "-out")
     },
     nativeWorkdir := {
-      val workdir = crossTarget.value / "native"
+      val workdir =
+        crossTarget.value / "native" / nativeConfig.value.targetTriple
```

Comment:
Do we need to compile into an added directory?

If you avoid this, will this remove the circular dependency?

Reply:
There is a caching mechanism to avoid recompiling the c and c++ files if they are already compiled.
It is not correct if you change the target triple. Because you need to invalidate the cache.
This moves the compiled files to separate directories so we can maintain the caching mechanism and avoid mixing them up.

Reply:
I am not aware of any caching in the plugin for compiling C or C++. 

Anyway, I was commenting about this piece of code due to the circular dependency you removed by not doing discovery. I would be more comfortable if the system would discover the target triple if it was not set based on the current "probe" vs the `clang` reported target triple.